astronomy/orbital_elements_body.hpp

```diff
@@ -0,0 +1,355 @@
+  return nodal_precession_;
+}
+
+inline OrbitalElements::Interval<Length>
+OrbitalElements::mean_semimajor_axis_range() const {
+  return mean_semimajor_axis_range_;
+}
+
+inline OrbitalElements::Interval<double>
+OrbitalElements::mean_eccentricity_range() const {
+  return mean_eccentricity_range_;
+}
+
+inline OrbitalElements::Interval<Angle>
+OrbitalElements::mean_inclination_range() const {
+  return mean_inclination_range_;
+}
+
+inline OrbitalElements::Interval<Angle>
+OrbitalElements::mean_longitude_of_ascending_node_range() const {
+  return mean_longitude_of_ascending_node_range_;
+}
+
+inline OrbitalElements::Interval<Angle>
+OrbitalElements::mean_argument_of_periapsis_range() const {
+  return mean_argument_of_periapsis_range_;
+}
+
+template<typename PrimaryCentred>
+std::vector<OrbitalElements::EquinoctialElements>
+OrbitalElements::OsculatingEquinoctialElements(
+    DiscreteTrajectory<PrimaryCentred> const& trajectory,
+    MassiveBody const& primary,
+    Body const& secondary) {
+  DegreesOfFreedom<PrimaryCentred> const primary_dof{
+      PrimaryCentred::origin, Velocity<PrimaryCentred>{}};
+  std::vector<EquinoctialElements> result;
+  for (auto it = trajectory.Begin(); it != trajectory.End(); ++it) {
+    auto const osculating_elements =
+        KeplerOrbit<PrimaryCentred>(primary,
+                                    secondary,
+                                    it.degrees_of_freedom() - primary_dof,
+                                    it.time())
+            .elements_at_epoch();
+    double const& e = *osculating_elements.eccentricity;
+    Angle const& ϖ = *osculating_elements.longitude_of_periapsis;
+    Angle const& Ω = osculating_elements.longitude_of_ascending_node;
+    Angle const& M = *osculating_elements.mean_anomaly;
+    Angle const& i = osculating_elements.inclination;
+    double const tg_½i = Tan(i / 2);
+    double const cotg_½i = 1 / tg_½i;
+    result.push_back(
+        {/*.t = */ it.time(),
+         /*.a = */ *osculating_elements.semimajor_axis,
+         /*.h = */ e * Sin(ϖ),
+         /*.k = */ e * Cos(ϖ),
+         /*.λ = */ result.empty() ? ϖ + M : UnwindFrom(result.back().λ, ϖ + M),
+         /*.p = */ tg_½i * Sin(Ω),
+         /*.q = */ tg_½i * Cos(Ω),
+         /*.pʹ = */ cotg_½i * Sin(Ω),
+         /*.qʹ = */ cotg_½i * Cos(Ω)});
+  }
+  return result;
+}
+
+inline std::vector<OrbitalElements::EquinoctialElements> const&
+OrbitalElements::osculating_equinoctial_elements() const {
+  return osculating_equinoctial_elements_;
+}
+
+inline std::vector<OrbitalElements::EquinoctialElements> const&
+OrbitalElements::mean_equinoctial_elements() const {
+  return mean_equinoctial_elements_;
+}
+
+inline Time OrbitalElements::SiderealPeriod(
+    std::vector<EquinoctialElements> const& equinoctial_elements) {
+  Time const Δt =
+      equinoctial_elements.back().t - equinoctial_elements.front().t;
+  Instant const t0 = equinoctial_elements.front().t + Δt / 2;
+  Product<Angle, Square<Time>> ſ_λt_dt;
```

Comment:
What is this identifier?  Is this top-half integral?  Why not integral?

Reply:
`∫` is not allowed in identifiers; neither is `⌠` for that matter. `ſ` is a long s; much like we use `Σ` for `∑`, and `Δ` instead of `∆`, I am using `ſ` as a ſubſtitute for `∫`.